spack.yaml

```diff
@@ -3,40 +3,64 @@
 # It describes a set of packages to be installed, along with
 # configuration settings.
 spack:
-  definitions:
-  - compilers:
-    - intel@2021.10.0
-  - mpis:
-    - openmpi@4.1.5
-  - serial_packages:
-    - fortranxml@4.1.2
-  - packages:
-    - esmf@8.5.0+debug
-    - parallelio@2.6.2 build_type=RelWithDebInfo fflags='-qno-opt-dynamic-align -convert
-      big_endian -assume byterecl -ftz -traceback -assume realloc_lhs -fp-model source'
-      cflags='-qno-opt-dynamic-align -fp-model precise -std=gnu99'
-    - netcdf-c@4.9.2 build_system=cmake build_type=RelWithDebInfo
-    - netcdf-fortran@4.6.1
-    - fms@2023.02 precision=64 +large_file +deprecated_io ~gfs_phys ~openmp ~quad_precision
-      build_type=RelWithDebInfo
-    - access-om3 +install_libraries
   specs:
-  - matrix:
-    - [$mpis]
-    - [$%compilers]
-  - matrix:
-    - [$packages]
-    - [$^mpis]
-    - [$%compilers]
-  - matrix:
-    - [$serial_packages]
-    - [$%compilers]
+    - access-om3@git.2024.04.0
+  packages:
+    access-om3-nuopc:
+      require:
+        - '@git.0.3.0'
+        - +install_libraries
+
+    esmf:
+      require:
+        - '@8.5.0'
+        - +debug
+    parallelio:
+      require:
+        - '@2.6.2'
+        - build_type=RelWithDebInfo fflags='-qno-opt-dynamic-align -convert big_endian -assume byterecl -ftz -traceback -assume realloc_lhs -fp-model source' cflags='-qno-opt-dynamic-align -fp-model precise -std=gnu99'
```

Comment:
If these build flags are never changed I wonder if they wouldn't be better defined in [`access-om3-nuopc`](https://github.com/ACCESS-NRI/spack-packages/blob/main/packages/access-om3-nuopc/package.py).

I believe one of the purposes of the SBD was that it would be possible for someone to `spack install access-om3` and it should work, and that this environment `spack.yaml` was for deployment and *specifying exact versions* where appropriate. If those flags are only specified here then I doubt the `spack install access-om3` would work as expected.

I'm not so sure about `build_type`. We could make a variant and wrap the logic of choosing and specifying the `build_type` arguments in a variant choice.

Reply:
Any thoughts @harshula?

Reply:
I'd be happy for these to be moved into the associated SPD. Would specifying them in the `spack.yaml` override the SPDs variants? I'll defer to @harshula on this one.